aws/resources/ec2_network_acl_test.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/stretchr/testify/require"
+)
+
+type mockedNetworkACL struct {
+	BaseAwsResource
+	ec2iface.EC2API
+
+	DescribeNetworkAclsOutput          ec2.DescribeNetworkAclsOutput
+	DeleteNetworkAclOutput             ec2.DeleteNetworkAclOutput
+	ReplaceNetworkAclAssociationOutput ec2.ReplaceNetworkAclAssociationOutput
+}
+
+func (m mockedNetworkACL) DescribeNetworkAcls(*ec2.DescribeNetworkAclsInput) (*ec2.DescribeNetworkAclsOutput, error) {
+	return &m.DescribeNetworkAclsOutput, nil
+}
+
+func (m mockedNetworkACL) DeleteNetworkAcl(*ec2.DeleteNetworkAclInput) (*ec2.DeleteNetworkAclOutput, error) {
+	return &m.DeleteNetworkAclOutput, nil
+}
+
+func (m mockedNetworkACL) ReplaceNetworkAclAssociation(*ec2.ReplaceNetworkAclAssociationInput) (*ec2.ReplaceNetworkAclAssociationOutput, error) {
+	return &m.ReplaceNetworkAclAssociationOutput, nil
+}
+
+func TestNetworkAcl_GetAll(t *testing.T) {
+	telemetry.InitTelemetry("cloud-nuke", "")
+
+	var (
+		now     = time.Now()
+		testId1 = "acl-09e36c45cbdbfb001"
+		testId2 = "acl-09e36c45cbdbfb002"
+
+		testName1 = "cloud-nuke-acl-001"
+		testName2 = "cloud-nuke-acl-002"
+	)
+
+	resourceObject := NetworkACL{
+		Client: mockedNetworkACL{
+			DescribeNetworkAclsOutput: ec2.DescribeNetworkAclsOutput{
+				NetworkAcls: []*ec2.NetworkAcl{
+					{
+						NetworkAclId: aws.String(testId1),
+						Tags: []*ec2.Tag{
+							{
+								Key:   aws.String("Name"),
+								Value: aws.String(testName1),
+							},
+							{
+								Key:   aws.String(util.FirstSeenTagKey),
+								Value: aws.String(util.FormatTimestamp(now)),
+							},
+						},
+					},
+					{
+						NetworkAclId: aws.String(testId2),
+						Tags: []*ec2.Tag{
+							{
+								Key:   aws.String("Name"),
+								Value: aws.String(testName2),
+							},
+							{
+								Key:   aws.String(util.FirstSeenTagKey),
+								Value: aws.String(util.FormatTimestamp(now.Add(1 * time.Hour))),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	resourceObject.BaseAwsResource.Init(nil)
+
+	tests := map[string]struct {
+		configObj config.ResourceType
+		expected  []string
+	}{
+		"emptyFilter": {
+			configObj: config.ResourceType{},
+			expected:  []string{testId1, testId2},
+		},
+		"nameExclusionFilter": {
+			configObj: config.ResourceType{
+				ExcludeRule: config.FilterRule{
+					NamesRegExp: []config.Expression{{
+						RE: *regexp.MustCompile(testName1),
+					}}},
+			},
+			expected: []string{testId2},
+		},
+		"nameInclusionFilter": {
+			configObj: config.ResourceType{
+				IncludeRule: config.FilterRule{
+					NamesRegExp: []config.Expression{{
+						RE: *regexp.MustCompile(testName1),
+					}}},
+			},
+			expected: []string{testId1},
+		},
+		"timeAfterExclusionFilter": {
+			configObj: config.ResourceType{
+				ExcludeRule: config.FilterRule{
+					TimeAfter: aws.Time(now),
+				}},
+			expected: []string{
+				testId1,
+			},
+		},
+	}
+	for name, tc := range tests {
```

Comment:
tc is not reinitialized, in test case, I suspect all tests will run with the same tc instance

Reply:
Not really sure about this comment. Can you elaborate and what you are suggesting?

Reply:
I wonder if we should revisit this at a follow-up PR. I can look into whether removing `t.Parallel()` fixes the problem and explore other option if necessary.